unstable test TestGlobalMemoryTuner

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
==================== Test output for //util/gctuner:gctuner_test (shard 3 of 5):

=== RUN   TestGlobalMemoryTuner

    memory_limit_tuner_test.go:97: 

        	Error Trace:	util/gctuner/memory_limit_tuner_test.go:97

        	Error:      	Should be true

        	Test:       	TestGlobalMemoryTuner

    memory_limit_tuner_test.go:71: 

        	Error Trace:	util/gctuner/memory_limit_tuner_test.go:71

        	            				GOROOT/src/runtime/panic.go:540

        	            				GOROOT/src/testing/testing.go:980

        	            				util/gctuner/memory_limit_tuner_test.go:97

        	Error:      	Condition never satisfied

        	Test:       	TestGlobalMemoryTuner

--- FAIL: TestGlobalMemoryTuner (6.14s)

```
### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/util/gctuner/memory_limit_tuner_test.go
+++ b/util/gctuner/memory_limit_tuner_test.go
@@ -97,7 +97,6 @@ func TestGlobalMemoryTuner(t *testing.T) {
 		return GlobalMemoryLimitTuner.waitingReset.Load() && gcNum < getNowGCNum()
 	}, 5*time.Second, 100*time.Millisecond)
 	// Test waiting for reset
-	time.Sleep(500 * time.Millisecond)
 	require.Eventually(t, func() bool {
 		return GlobalMemoryLimitTuner.calcMemoryLimit(fallbackPercentage) == debug.SetMemoryLimit(-1)
 	}, 5*time.Second, 100*time.Millisecond)
@@ -111,7 +110,6 @@ func TestGlobalMemoryTuner(t *testing.T) {
 	allocator.free(memory100mb)
 	runtime.GC()
 	// Trigger GC in 80% again
-	time.Sleep(500 * time.Millisecond)
 	require.Eventually(t, func() bool {
 		return GlobalMemoryLimitTuner.calcMemoryLimit(GlobalMemoryLimitTuner.GetPercentage()) == debug.SetMemoryLimit(-1)
 	}, 5*time.Second, 100*time.Millisecond)
@@ -119,7 +117,6 @@ func TestGlobalMemoryTuner(t *testing.T) {
 	gcNum = getNowGCNum()
 	checkNextGCEqualMemoryLimit()
 	memory210mb = allocator.alloc(210 << 20)
-	time.Sleep(100 * time.Millisecond)
 	require.Eventually(t, func() bool {
 		return gcNum < getNowGCNum()
 	}, 5*time.Second, 100*time.Millisecond)

--- a/util/gctuner/memory_limit_tuner_test.go
+++ b/util/gctuner/memory_limit_tuner_test.go
@@ -93,29 +93,36 @@ func TestGlobalMemoryTuner(t *testing.T) {
 	gcNum := getNowGCNum()
 
 	memory210mb := allocator.alloc(210 << 20)
-	time.Sleep(100 * time.Millisecond)
-	require.True(t, GlobalMemoryLimitTuner.waitingReset.Load())
-	require.True(t, gcNum < getNowGCNum())
+	require.Eventually(t, func() bool {
+		return GlobalMemoryLimitTuner.waitingReset.Load() && gcNum < getNowGCNum()
+	}, 5*time.Second, 100*time.Millisecond)
 	// Test waiting for reset
 	time.Sleep(500 * time.Millisecond)
-	require.Equal(t, GlobalMemoryLimitTuner.calcMemoryLimit(fallbackPercentage), debug.SetMemoryLimit(-1))
+	require.Eventually(t, func() bool {
+		return GlobalMemoryLimitTuner.calcMemoryLimit(fallbackPercentage) == debug.SetMemoryLimit(-1)
+	}, 5*time.Second, 100*time.Millisecond)
 	gcNum = getNowGCNum()
 	memory100mb := allocator.alloc(100 << 20)
-	time.Sleep(100 * time.Millisecond)
-	require.Equal(t, gcNum, getNowGCNum()) // No GC
+	require.Eventually(t, func() bool {
+		return gcNum == getNowGCNum()
+	}, 5*time.Second, 100*time.Millisecond) // No GC
 
 	allocator.free(memory210mb)
 	allocator.free(memory100mb)
 	runtime.GC()
 	// Trigger GC in 80% again
 	time.Sleep(500 * time.Millisecond)
-	require.Equal(t, GlobalMemoryLimitTuner.calcMemoryLimit(GlobalMemoryLimitTuner.GetPercentage()), debug.SetMemoryLimit(-1))
+	require.Eventually(t, func() bool {
+		return GlobalMemoryLimitTuner.calcMemoryLimit(GlobalMemoryLimitTuner.GetPercentage()) == debug.SetMemoryLimit(-1)
+	}, 5*time.Second, 100*time.Millisecond)
 	time.Sleep(100 * time.Millisecond)
 	gcNum = getNowGCNum()
 	checkNextGCEqualMemoryLimit()
 	memory210mb = allocator.alloc(210 << 20)
 	time.Sleep(100 * time.Millisecond)
-	require.True(t, gcNum < getNowGCNum())
+	require.Eventually(t, func() bool {
+		return gcNum < getNowGCNum()
+	}, 5*time.Second, 100*time.Millisecond)
 	allocator.free(memory210mb)
 	allocator.free(memory600mb)
 }

